src/ui/ResultsPerPage/ResultsPerPage.ts

```diff
@@ -164,6 +170,12 @@ export class ResultsPerPage extends Component {
     );
   }
 
+  private resolveInitialState() {
+    this.currentResultsPerPage = this.getInitialChoice();
```

Comment:
I recently worked with this component. I wanted to point out this method which updates this property as well as the property on the search interface:

```
private updateResultsPerPage(resultsPerPage: number) {
    this.searchInterface.resultsPerPage = resultsPerPage;
    this.currentResultsPerPage = resultsPerPage;
}
```

Also, why is the